pkg/controller/postgresql.go

```diff
@@ -146,86 +145,112 @@ func (c *Controller) processEvent(obj interface{}) error {
-		c.stopChs[clusterName] = stopCh
-		c.clustersMu.Unlock()
+		func() {
+			defer c.clustersMu.Unlock()
+			c.clustersMu.Lock()
+
+			c.teamClusters[teamName] = append(c.teamClusters[teamName], clusterName)
+			c.clusters[clusterName] = cl
+			c.stopChs[clusterName] = stopCh
+			c.clusterLogs[clusterName] = ringlog.New(c.opConfig.ClusterLogSize)
+		}()
 
 		if err := cl.Create(); err != nil {
 			cl.Error = fmt.Errorf("could not create cluster: %v", err)
-			logger.Errorf("%v", cl.Error)
+			lg.Errorf("%v", cl.Error)
 
 			return nil
 		}
 
-		logger.Infof("Cluster %q has been created", clusterName)
+		lg.Infoln("Cluster has been created")
 	case spec.EventUpdate:
-		logger.Infof("Update of the %q cluster started", clusterName)
+		lg.Infoln("Update of the cluster started")
 
 		if !clusterFound {
-			logger.Warnf("Cluster %q does not exist", clusterName)
+			lg.Warnln("Cluster does not exist")
 			return nil
 		}
 		if err := cl.Update(event.NewSpec); err != nil {
 			cl.Error = fmt.Errorf("could not update cluster: %v", err)
-			logger.Errorf("%v", cl.Error)
+			lg.Errorf("%v", cl.Error)
 
 			return nil
 		}
 		cl.Error = nil
-		logger.Infof("Cluster %q has been updated", clusterName)
+		lg.Infoln("Cluster has been updated")
 	case spec.EventDelete:
-		logger.Infof("Deletion of the %q cluster started", clusterName)
+		teamName := strings.ToLower(cl.Spec.TeamID)
+
+		lg.Infoln("Deletion of the cluster started")
 		if !clusterFound {
-			logger.Errorf("Unknown cluster: %q", clusterName)
+			lg.Errorln("Unknown cluster")
 			return nil
 		}
 
 		if err := cl.Delete(); err != nil {
-			logger.Errorf("could not delete cluster %q: %v", clusterName, err)
+			lg.Errorf("could not delete cluster: %v", err)
 			return nil
 		}
 		close(c.stopChs[clusterName])
 
-		c.clustersMu.Lock()
-		delete(c.clusters, clusterName)
-		delete(c.stopChs, clusterName)
-		c.clustersMu.Unlock()
+		func() {
+			defer c.clustersMu.Unlock()
+			c.clustersMu.Lock()
+
+			delete(c.clusters, clusterName)
+			delete(c.stopChs, clusterName)
+			delete(c.clusterLogs, clusterName)
+			for i, val := range c.teamClusters[teamName] { // on relativel
+				if val == clusterName {
+					copy(c.teamClusters[teamName][i:], c.teamClusters[teamName][i+1:])
+					c.teamClusters[teamName][len(c.teamClusters[teamName])-1] = spec.NamespacedName{}
+					c.teamClusters[teamName] = c.teamClusters[teamName][:len(c.teamClusters[teamName])-1]
+					break
+				}
+			}
+		}()
 
-		logger.Infof("Cluster %q has been deleted", clusterName)
+		lg.Infoln("Cluster has been deleted")
 	case spec.EventSync:
-		logger.Infof("Syncing of the %q cluster started", clusterName)
+		lg.Infoln("Syncing of the cluster started")
 
 		// no race condition because a cluster is always processed by single worker
 		if !clusterFound {
 			stopCh := make(chan struct{})
-			cl = cluster.New(c.makeClusterConfig(), c.KubeClient, *event.NewSpec, logger)
+			cl = cluster.New(c.makeClusterConfig(), c.KubeClient, *event.NewSpec, lg)
```

Comment:
that parts duplicates the code from Create, perhaps refactor it into a function?